src/string/palindrome.rs

```diff
@@ -1,11 +1,5 @@
 pub fn is_palindrome(s: &str) -> bool {
-    let mut chars = s.chars();
-    while let (Some(c1), Some(c2)) = (chars.next(), chars.next_back()) {
-        if c1 != c2 {
-            return false;
-        }
-    }
-    true
+    s.chars().eq(s.chars().rev())
```

Comment:
This is a nice one liner but quite inefficient (both in terms of space and time). Would you like to rewrite it avowing the copy and avoiding the redundant checks in case of a palindrome?

Reply:
Agree with you, at this point it is better to close the PR